src/internal/packager/helm/post-render.go

```diff
@@ -33,16 +35,24 @@ type renderer struct {
 func (h *Helm) newRenderer() (*renderer, error) {
 	message.Debugf("helm.NewRenderer()")
 
-	namespaces := make(map[string]*corev1.Namespace)
-	if h.cluster != nil {
-		namespaces[h.chart.Namespace] = h.cluster.NewZarfManagedNamespace(h.chart.Namespace)
-	}
-
-	return &renderer{
+	rend := &renderer{
 		Helm:           h,
-		connectStrings: make(types.ConnectStrings),
-		namespaces:     namespaces,
-	}, nil
+		connectStrings: types.ConnectStrings{},
+		namespaces:     map[string]*corev1.Namespace{},
+	}
+	if h.cluster == nil {
+		return rend, nil
+	}
+	namespace, err := h.cluster.K8s.GetNamespace(h.chart.Namespace)
+	if err != nil && !errors.IsNotFound(err) {
+		return nil, fmt.Errorf("unable to check for existing namespace %q in cluster: %w", h.chart.Namespace, err)
+	}
+	if h.cfg.DeployOpts.AdoptExistingResources && !errors.IsNotFound(err) {
```

Comment:
Do we need the `!errors.IsNotFound` again here?
Also I would switch the statement to be like
```
if error.IsNotFound {
 NewZarfManagedNamespace
else {
UpdateNamespaceToBeZarfManaged
}
```

Reply:
AFAIK there's no way around it. First check is checking if there is a true err instead of the namespace not being found. Second check if to see if the namespace doesn't already exist. 

Thanks for the rec, reads nicer without the double negative